File: algorithm_of_notes/java/com/algorithm/sort/quick/Quick.java
package com.algorithm.sort.quick;

import java.util.Arrays;

/**
 * 快速排序
 * Created by 余林丰 on 2017/6/26/0026.
 */
public class Quick {
    public static void main(String[] args) {
        int[] nums = {6, 1, 2, 7, 9, 3, 4, 5, 10, 8};
        nums = quickSort(nums, 0, nums.length - 1);
        System.out.println(Arrays.toString(nums));
    }
    
    /**
     * 快速排序
     * @param nums 待排序数组序列
     * @param left 数组第一个元素索引
     * @param right 数组最后一个元素索引
     * @return 排好序的数组序列
     */
    private static int[] quickSort(int[] nums, int left, int right) {
        if (left < right) {
            int temp = nums[left];
            int i = left;
            int j = right;
            while (i < j) {
                while (i < j && nums[j] >= temp) {
                    j--;
                }
                if (i < j) {
                    nums[i] = nums[j];
                    i++;
                }
                while (i < j && nums[i] < temp) {
                    i++;
                }
                while (i < j) {
                    nums[j] = nums[i];
                    j--;
                }
            }
            nums[i] = temp;
            quickSort(nums, left, i - 1);
            quickSort(nums, i + 1, right);
        }
        return nums;
    }
}

File: algorithm_of_notes/python/sort/Quick.py
#快速排序
def quick_sort(nums, left, right):
    if left < right:
        temp = nums[left]
        i = left
        j = right
        while i < j:
            while i < j and nums[j] >= temp:
                j -= 1
            if i < j:
                nums[i] = nums[j]
                i += 1
            while i < j and nums[i] < temp:
                i += 1
            if i < j:
                nums[j] = nums[i]
                j -= 1
        nums[i] = temp
        quick_sort(nums, left, i - 1)
        quick_sort(nums, i + 1, right)
    
    return nums

nums = [6, 1, 2, 7, 9, 3, 4, 5, 10, 8]
nums = quick_sort(nums, 0, len(nums) - 1)
print(nums)